### PR TITLE
Pool lens integration

### PR DESCRIPTION
**Pull request by defistar**
# Change Details

1. PoolLens contract gets all details of a Pool by comptrollerAddress and PoolId
2. In isolated-pools subgraph, poolDetails are to be queried while processing PoolRegistered Event
3. PoolLens config, ABI are added to the subgraph: isolated-pools
4. PoolLens integration point is in `subgraphs/isolated-pools/src/operations/create.ts`
5. mocking for PoolLens integration is under: `tests/VToken/mocks.ts`

### CODE REVIEW

File: subgraphs/isolated-pools/src/operations/create.ts [hunk 1]
```diff
@@ -34,18 +36,26 @@ import { getTransactionEventId } from '../utilities/ids';
...
-  pool.priceOracle = Address.fromString('0x0000000000000000000000000000000000000000');
-  pool.pauseGuardian = Address.fromString('0x0000000000000000000000000000000000000000');
-  pool.closeFactor = new BigInt(0);
-  pool.liquidationIncentive = new BigInt(0);
+  pool.riskRating = poolDataFromLens.riskRating.toString();
+  pool.category = poolDataFromLens.category;
+  pool.logoURL = poolDataFromLens.logoURL;
+  pool.description = poolDataFromLens.description;
+  pool.priceOracle = poolDataFromLens.priceOracle;
+  pool.pauseGuardian = pauseGuardianAddress;
```
Comment (coreyar) [inline subgraphs/isolated-pools/src/operations/create.ts:53]: I think we can remove this. It was removed from the `getPoolByComptroller`

File: subgraphs/isolated-pools/template.yaml [hunk 2]
```diff
@@ -30,11 +30,13 @@ dataSources:
...
         - name: BEP20
           file: ../../packages/isolated-pools-abis/Bep20.json
+        - name: PoolLens
+          file: ../../packages/isolated-pools-abis/PoolLens.json  
       eventHandlers:
         - event: PoolRegistered(uint256,(uint256,string,address,address,uint256,uint256))
           handler: handlePoolRegistered
         - event: PoolNameSet(uint256,string)
-          handler: handlePoolNameSet
+          handler: handlePoolNameSet
```
Comment (coreyar) [inline subgraphs/isolated-pools/template.yaml:39]: extra space. Maybe we could use a yaml formatting tool?

File: subgraphs/isolated-pools/tests/PoolRegistry/index.test.ts [hunk 3]
```diff
@@ -21,7 +21,21 @@ beforeAll(() => {
...
+    ],
+    [
+      ethereum.Value.fromString('Gamer Pool'),
+      ethereum.Value.fromAddress(Address.fromString('0x0000000000000000000000000000000000000072')),
+      ethereum.Value.fromAddress(Address.fromString('0x0000000000000000000000000000000000000064')),
+      ethereum.Value.fromUnsignedBigInt(BigInt.fromI32(9000000)),
+      ethereum.Value.fromUnsignedBigInt(BigInt.fromI32(6235232)),
+    ],
+    [
+      ethereum.Value.fromString('Gamer Pool'),
```
Comment (coreyar) [inline subgraphs/isolated-pools/tests/PoolRegistry/index.test.ts:36]: For clarity we could give these different names

Comment (coreyar) [reply subgraphs/isolated-pools/tests/PoolRegistry/index.test.ts:36]: We also need to update the test

Comment (defistar) [reply subgraphs/isolated-pools/tests/PoolRegistry/index.test.ts:36]: @coreyar different names in the test where we are creating mock?

like `Gamer-1 Pool` , `Gamer-2 Pool`, `Gamer-3 Pool`, `Gamer-4 Pool`

File: subgraphs/isolated-pools/template.yaml (hunk 2 above)
Comment (defistar) [reply subgraphs/isolated-pools/template.yaml:39]: @coreyar 
yeah need to add formatting tool

File: subgraphs/isolated-pools/src/operations/create.ts (hunk 1 above)
Comment (defistar) [reply subgraphs/isolated-pools/src/operations/create.ts:53]: @coreyar 
> I think we can remove this. It was removed from the getPoolByComptroller
are you refering to property `pauseGuardian` ?

this is yet to be removed from contract. i will do it now

File: subgraphs/isolated-pools/tests/PoolRegistry/index.test.ts [hunk 4]
```diff
@@ -90,6 +90,6 @@ describe('Pool Registry', () => {
     logStore();
     handlePoolNameSet(poolNameSetEvent);
     logStore();
-    assertPoolDocument('name', 'Summer Pool');
+    assertPoolDocument('name', 'Gamer Pool2');
```
Comment (coreyar) [inline subgraphs/isolated-pools/tests/PoolRegistry/index.test.ts:93]: This is not correct. This is the original name of the pool

File: packages/isolated-pools-abis/CToken.json [hunk 5]
```diff
@@ -845,19 +896,17 @@
     "type": "function"
   },
   {
-    "constant": true,
     "inputs": [],
-    "name": "underlying",
```
Comment (coreyar) [inline packages/isolated-pools-abis/CToken.json:850]: The underlying property shouldn't be removed

File: packages/isolated-pools-abis/CToken.json [hunk 6]
```diff
@@ -366,7 +366,7 @@
       {
         "indexed": false,
         "internalType": "address",
-        "name": "cTokenCollateral",
+        "name": "vTokenCollateral",
```
[the remark this answers is not in the document]
Comment (coreyar) [reply packages/isolated-pools-abis/CToken.json:369]: This file should be removed in favor or VToken

**Review by coreyar — APPROVED** [review]

**Pull request merged**